File: ws-gateway/scripts/test-redis-subscribe.ts
import { createClient } from "redis";

const REDIS_URL = "redis://localhost:6379";

async function main() {
  const subscriber = createClient({ url: REDIS_URL });
  await subscriber.connect();

  await subscriber.subscribe("interview:test", (message) => {
    console.log("Mensaje recibido:", message);
  });

  console.log("Escuchando el canal 'interview:test'...");
}

main();

File: backend/scripts/test_redis_subscribe.py
import os

import redis
from dotenv import load_dotenv

load_dotenv()

r = redis.from_url(os.environ.get("REDIS_URL"))
pubsub = r.pubsub()
pubsub.subscribe("interview:test")

print("Escuchando el canal 'interview:test'...")
for message in pubsub.listen():
    if message["type"] == "message":
        print("Mensaje recibido:", message["data"].decode("utf-8"))